src/dotnet/APIView/APIViewWeb/Pages/Assemblies/Index.cshtml.cs

```diff
@@ -39,6 +41,13 @@ public IndexPageModel(ReviewManager manager, UserPreferenceCache preferenceCache
             IEnumerable<string> search, IEnumerable<string> languages, IEnumerable<string> state,
             IEnumerable<string> status, IEnumerable<string> type, int pageNo=1, int pageSize=_defaultPageSize, string sortField=_defaultSortField)
         {
+            var userProfile = await _userProfileManager.tryGetUserProfileAsync(User);
+
+            if(User.Identity.IsAuthenticated && (userProfile.UserName == null || userProfile.Email == null))
```

Comment:
I am worried about this user experience change. @JonathanGiles We will force all users to update their email address when they first login after this change is deployed. I am worried that many won't be happy with that change. Can we fetch email address from GitHub profile and create it as default profile?

Reply:
Would it be such a problem? Seeing this is a one time thing.

Reply:
Can we make an exception for test authentication? https://github.com/Azure/azure-sdk-tools/blob/b009cdc5e2dcaf73d3d9020f4aca1a2a269ad700/src/dotnet/APIView/APIViewWeb/Startup.cs#L108

Reply:
This is no longer as impactful. There is no forced redirect or email requirement, and more information is added about them.